cp_diffdock_protocol.py

```diff
@@ -0,0 +1,121 @@
+import json
+from dataclasses import asdict, astuple, dataclass, field
+from typing import List, Any
+from collections import namedtuple
+
+class DiffDockProtocol:
+    # Can be used like enum
+    MessageType = namedtuple('MessageType', ['INVALID', 'ERROR', 'RESULTS', 'STATUS'])('invalid', 'error', 'results', 'status')
```

Comment:
I like this style of more rigorous use of enumerated strings, rather than just strings (which I need to start using). I guess here it's not python's Enum because strings interoperate with other languages more understandably than the ints that underlie Enum.

Reply:
The c++ json package has a macro for conveniently serializing enums as strings!